### PR TITLE
v3.0.0 Run client in pod

### PR DESCRIPTION
**Pull request by caoxianfei1**
This pr packages all client binary to the mirror.

### CODE REVIEW

File: deploy/manifests/node-plugin-daemonset.yaml [hunk 1]
```diff
@@ -69,6 +69,8 @@ spec:
               fieldPath: metadata.namespace
         - name: CSI_ENDPOINT
           value: unix:///csi/csi.sock
+        - name: MDSADDR
+          value: 59.111.93.76:23970,59.111.93.77:23970,59.111.93.78:23970
```
Comment (h0hmj) [inline deploy/manifests/node-plugin-daemonset.yaml:73]: replace ips with fake ones

File: deploy/manifests/node-plugin-daemonset.yaml [hunk 2]
```diff
@@ -21,7 +21,7 @@ spec:
       dnsPolicy: ClusterFirstWithHostNet
       containers:
       - name: driver-registrar
-        image: registry.k8s.io/sig-storage/csi-node-driver-registrar:v2.2.0
+        image: harbor.cloud.netease.com/curve/csi-node-driver-registrar:v2.2.0
```
Comment (h0hmj) [inline deploy/manifests/node-plugin-daemonset.yaml:24]: necessary to change registry.k8s.io/sig-storage images?

File: examples/pod-block.yaml [hunk 3]
```diff
@@ -11,7 +11,7 @@ spec:
     image: debian
     command:
     - sleep
-    - "3600"
+    - "36000"
```
Comment (h0hmj) [inline examples/pod-block.yaml:14]: necessary to change?

File: examples/pvc-block.yaml [hunk 4]
```diff
@@ -9,4 +9,4 @@ spec:
   storageClassName: curve
   resources:
     requests:
-      storage: 20Gi
+      storage: 10Gi
```
Comment (h0hmj) [inline examples/pvc-block.yaml:12]: maybe add a hint of minimal size (10G)

**Review by h0hmj — CHANGES_REQUESTED** [review]

File: build/curve-csi/Dockerfile [hunk 5]
```diff
@@ -14,8 +19,43 @@ LABEL name="curve-csi" \
...
+
+RUN apt-get update -y && \
+    apt-get install -y coreutils dnsutils iputils-ping iproute2 telnet curl vim less wget graphviz unzip tcpdump gdb daemon procps python && \
+    apt-get install -y ca-certificates e2fsprogs mount xfsprogs udev && \
+    apt-get clean
+
+COPY ./build/curve-csi/entrypoint.sh /root/
+
+# Get and install curve, curve-nbd and nebd
+RUN wget https://pkg.cloud.netease.com/upload/$/TtewH -O curve-sdk_1.2.6+72cb7582_amd64.deb && \
```
[the remark this answers is not in the document]
Comment (caoxianfei1) [reply build/curve-csi/Dockerfile]: done

File: deploy/manifests/node-plugin-daemonset.yaml (hunk 1 above)
Comment (caoxianfei1) [reply deploy/manifests/node-plugin-daemonset.yaml]: done

File: deploy/manifests/node-plugin-daemonset.yaml (hunk 2 above)
Comment (caoxianfei1) [reply deploy/manifests/node-plugin-daemonset.yaml]: done

File: examples/pod-block.yaml (hunk 3 above)
Comment (caoxianfei1) [reply examples/pod-block.yaml]: done

File: examples/pvc-block.yaml (hunk 4 above)
Comment (caoxianfei1) [reply examples/pvc-block.yaml]: done

**Review by h0hmj — APPROVED** [review]

**Pull request merged**